### PR TITLE
This improves slightly the documentation, adding instructions for CMake users.

### PR DESCRIPTION
**Pull request by lemire**
I think we should help cmake users a bit.

### CODE REVIEW

**Comment by lemire** [comment]
@jkeiser In the future, we may set SIMDJSON_JUST_LIBRARY to ON by default to ease CMake integration, but I don't want to this for 0.4 as we want 0.4 to come out earlier rather than later.

**Comment by jkeiser** [comment]
@lemire agreed. If we can't detect it (and perhaps we just shouldn't!) we should just add a cmake -DEV=1 or something (preferably short) so that we can easily get all of our stuff without inconveniencing others.

File: doc/basics.md [hunk 1]
```diff
@@ -168,6 +197,38 @@ And another one:
   cout << "number: " << v << endl;
 ```
 
+C++17 Support
```
Comment (jkeiser) [inline doc/basics.md:200]: Pure curiosity (I have absolutely no skin in the game): why did you move this?

File: doc/basics.md [hunk 2]
```diff
@@ -44,6 +48,31 @@ Note:
 - Visual Studio users should compile with the `_CRT_SECURE_NO_WARNINGS` flag to avoid warnings with respect to our use of standard C functions such as `fopen`.
 
 
+
+
+Using simdjson as a CMake dependency
+------------------
+
+
```
Comment (jkeiser) [inline doc/basics.md:56]: Can you remove these two newlines? And the two extras above the title? (Alternatively, we can keep all the distances the same.)

I'm so so sorry for this comment. It'll bug me forever though.

File: doc/basics.md [hunk 3]
```diff
@@ -44,6 +48,31 @@ Note:
...
+```
+set(SIMDJSON_JUST_LIBRARY ON CACHE STRING "Build just the library, nothing else." FORCE)
+add_subdirectory(simdjson EXCLUDE_FROM_ALL)
+```
+
+Elsewhere in your project, you can  declare dependencies on simdjson with lines such as these:
+
+```
+add_executable(myprogram myprogram.cpp)
+
```
Comment (jkeiser) [inline doc/basics.md:70]: Don't think this newline helps either, and it might actually affect readability a tiny bit by making users think the pause matters.

File: doc/basics.md (hunk 2 above)
Comment (lemire) [reply doc/basics.md:56]: Done.

File: doc/basics.md (hunk 3 above)
Comment (lemire) [reply doc/basics.md:70]: Done.

File: doc/basics.md (hunk 1 above)
Comment (lemire) [reply doc/basics.md:200]: Because it came after "Minifying JSON strings without parsing" but was logically meant to follow "Using the Parsed JSON".

I am fixing my own mistake with this change, btw.

**Comment by lemire** [comment]
I will merge. I tried to make the blank lines add up. :-)
